inlang/source-code/editor/src/pages/@host/@owner/@repository/components/Gitfloat.tsx

```diff
@@ -291,7 +316,35 @@ export const Gitfloat = () => {
 						lixErrors().length === 0
 					}
 				>
-					<div class="w-full flex justify-start items-center rounded-lg bg-inverted-surface shadow-xl ">
+					<Show when={forkStatus() && forkStatus().behind > 0 && !forkStatus().conflicts && userIsCollaborator()}>
+						<div class="syncfork w-full relative flex justify-start items-center gap-1.5 rounded-t-lg bg-[#293344] p-1.5 text-xs font-medium text-on-inverted-surface after:content-[''] after:absolute after:w-full after:h-8 after:translate-y-6 after:-m-1.5 after:bg-[#293344] after:-z-10">
+							<div class="text-warning-on-inverted-container px-1.5 py-[5px]">
+								<WarningIcon />
+							</div>
+							Get new project changes
+							<sl-button
+								prop:size="small"
+								onClick={async () => {
+									setIsMerging(true)
+									await repo()?.mergeUpstream()
+									refetchRepo()
+									setIsMerging(false)
+									setTimeout(() => {
+										mutateForkStatus({ ...forkStatus(), behind: 0, conflicts: false })
```

Comment:
i assume this is an optimisstic update?

Reply:
Yes, I can add the refetch of forkStatus(), but the value only changes once it's finished. Therefore, after the animation is finished, I mutate the value to the default.